src/Controller/Component/SearchComponent.php

```diff
@@ -103,8 +104,16 @@ public function beforeRender()
             if (method_exists($controller, 'fetchTable')) {
                 $method = 'fetchTable';
             }
+            $modelClassName = $this->getConfig('modelClass');
+            if (!$modelClassName && $method === 'fetchTable') {
```

Comment:
We only invoke this in this very specific instance to keep BC, nothing else should be affected, like happy normal path.